.github/workflows/continuous-integration-workflow.yml

```diff
@@ -0,0 +1,310 @@
+name: Continuous Integration Workflow
+
+on: [push]
+
+jobs:
+  go-postgres:
+    name: Go Postgres
+    runs-on: self-hosted
+    container: smartcontract/builder:1.0.31
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+          # Set health checks to wait until postgres has started
+          options: >-
+            --health-cmd pg_isready
+            --health-interval 10s
+            --health-timeout 5s
+            --health-retries 5
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-yarn
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Cache Go vendor packages
+        uses: actions/cache@v1
+        with:
+          path: /go/pkg/mod
+          key: $${{ secrets.CACHE_VERSION }}-go-mod-$${ hashFiles('go.sum') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION }}-go-mod
+      - name: Download Go vendor packages
+        run: go mod download
+      - name: Set up Yarn for Contracts
+        run: yarn setup:contracts
+      - name: Run tests
+        run: go test -v -p 6 -parallel 4 ./...
+        env:
+          DATABASE_URL: postgres://postgres:postgres@postgres:5432/postgres?sslmode=disable
+
+  rust:
+    name: Rust
+    runs-on: ubuntu-latest
+    container:
+      image: smartcontract/builder:1.0.31
+      env:
+        CARGO_HOME: /root/.cargo
+        RUSTUP_HOME: /root/.rustup
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run Rust test
+        run: ./tools/ci/rust_test
+
+  geth-postgres:
+    name: Geth Postgres
+    runs-on: self-hosted
+    env:
+      GETH_MODE: true
+      CI: true
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run tests
+        run: cd ./tools/docker && timeout --foreground 1200s ./compose test
+
+  parity-postgres:
+    name: Parity Postgres
+    runs-on: self-hosted
+    env:
+      CI: true
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run tests
+        run: cd ./tools/docker && timeout --foreground 1200s ./compose test
+
+  solidity:
+    name: Solidity
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
```

Comment:
I believe string interpolation for GHA only requires one `$`, as seen [here](https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions)